modules/drools/src/test/java/org.mule.module.drools/DroolsAssertBehaviorTestCase.java

```diff
@@ -0,0 +1,130 @@
+import static org.mule.tck.MuleTestUtils.testWithSystemProperty;
+import static org.mule.tck.MuleTestUtils.TestCallback;
+import static org.junit.runners.Parameterized.Parameters;
+
+
+@RunWith(Parameterized. class)
+public class DroolsAssertBehaviorTestCase extends AbstractMuleTestCase
+{
+
+    private final static Drools drools = new Drools();
+    private final static Rules rules = mock(Rules.class);
+    private final static MessageService messageService = mock(MessageService.class, RETURNS_DEEP_STUBS);
+    private final String propertyValue ;
+    private final TestCallback testCallback ;
+
+    public DroolsAssertBehaviorTestCase(String propertyValue, TestCallback testCallback) {
+        this.propertyValue = propertyValue;
+        this.testCallback = testCallback;
+    }
+
+    @Parameters
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(new Object[][]{
+                {"false", identityBehaviourCallback},
+                {"true", equalityBehaviourCallback}
+        });
+    }
+
+    private static void setUp () throws Exception {
+        when(rules.getResource()).thenReturn("rulesFile.drl");
+        drools.setMessageService(messageService);
+        DroolsSessionData sessionData = (DroolsSessionData) drools.createSession(rules);
+        when(rules.getSessionData()).thenReturn(sessionData);
+    }
+
+    @Test
+    public void testAssertBehaviour() throws Exception
+    {
+        testWithSystemProperty(USE_EQUALITY_ASSERT_BEHAVIOR, propertyValue, testCallback);
+    }
+
+    public static class TestFact {
+        private final String id;
+        private String description;
+
+        public TestFact(String id, String description) {
+            this.id = id;
+            this.description = description;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            TestFact testFact = (TestFact) o;
+
+            if (id != null ? !id.equals(testFact.id) : testFact.id != null) return false;
+            return description != null ? description.equals(testFact.description) : testFact.description == null;
+
+        }
+
+        @Override
+        public int hashCode() {
+            int result = id != null ? id.hashCode() : 0;
+            result = 31 * result + (description != null ? description.hashCode() : 0);
+            return result;
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public void setDescription(String description)
+        {
+            this.description = description;
+        }
+    }
+
+    static TestCallback identityBehaviourCallback =  new TestCallback(){
+        @Override
+        public void run() throws Exception {
+            setUp();
```

Comment:
create different methods then, one for the class setup and another one for the instance setup